ivy/functional/backends/tensorflow/experimental/statistical.py

```diff
@@ -663,12 +667,8 @@ def cummax(
     dtype: Optional[tf.DType] = None,
     out: Optional[Union[tf.Tensor, tf.Variable]] = None,
 ) -> Tuple[tf.Tensor, tf.Tensor]:
-    if x.dtype in (tf.bool, tf.float16):
```

Comment:
do all these `dtypes` work without any type casting?

Reply:
for d in ('float32', 'float64', 'int32', 'uint8', 'int16', 
          'int8', 'complex64', 'int64', 'bfloat16', 'uint16', 
          'complex128', 'float16', 'uint32', 'uint64'): 
    x = tf.constant([1, 2,], dtype=d)
    print(cummax(x))        
    
(<tf.Tensor: shape=(2,), dtype=float32, numpy=array([1., 2.], dtype=float32)>, <tf.Tensor: shape=(2,), dtype=int64, numpy=array([0, 1], dtype=int64)>)
(<tf.Tensor: shape=(2,), dtype=float64, numpy=array([1., 2.])>, <tf.Tensor: shape=(2,), dtype=int64, numpy=array([0, 1], dtype=int64)>)
(<tf.Tensor: shape=(2,), dtype=int32, numpy=array([1, 2])>, <tf.Tensor: shape=(2,), dtype=int64, numpy=array([0, 1], dtype=int64)>)
(<tf.Tensor: shape=(2,), dtype=uint8, numpy=array([1, 2], dtype=uint8)>, <tf.Tensor: shape=(2,), dtype=int64, numpy=array([0, 1], dtype=int64)>)
(<tf.Tensor: shape=(2,), dtype=int16, numpy=array([1, 2], dtype=int16)>, <tf.Tensor: shape=(2,), dtype=int64, numpy=array([0, 1], dtype=int64)>)
(<tf.Tensor: shape=(2,), dtype=int8, numpy=array([1, 2], dtype=int8)>, <tf.Tensor: shape=(2,), dtype=int64, numpy=array([0, 1], dtype=int64)>)
(<tf.Tensor: shape=(2,), dtype=float32, numpy=array([1., 2.], dtype=float32)>, <tf.Tensor: shape=(2,), dtype=int64, numpy=array([0, 1], dtype=int64)>)
(<tf.Tensor: shape=(2,), dtype=int64, numpy=array([1, 2], dtype=int64)>, <tf.Tensor: shape=(2,), dtype=int64, numpy=array([0, 1], dtype=int64)>)
(<tf.Tensor: shape=(2,), dtype=bfloat16, numpy=array([1, 2], dtype=bfloat16)>, <tf.Tensor: shape=(2,), dtype=int64, numpy=array([0, 1], dtype=int64)>)
(<tf.Tensor: shape=(2,), dtype=uint16, numpy=array([1, 2], dtype=uint16)>, <tf.Tensor: shape=(2,), dtype=int64, numpy=array([0, 1], dtype=int64)>)
(<tf.Tensor: shape=(2,), dtype=float64, numpy=array([1., 2.])>, <tf.Tensor: shape=(2,), dtype=int64, numpy=array([0, 1], dtype=int64)>)
(<tf.Tensor: shape=(2,), dtype=float16, numpy=array([1., 2.], dtype=float16)>, <tf.Tensor: shape=(2,), dtype=int64, numpy=array([0, 1], dtype=int64)>)
(<tf.Tensor: shape=(2,), dtype=uint32, numpy=array([1, 2], dtype=uint32)>, <tf.Tensor: shape=(2,), dtype=int64, numpy=array([0, 1], dtype=int64)>)
(<tf.Tensor: shape=(2,), dtype=uint64, numpy=array([1, 2], dtype=uint64)>, <tf.Tensor: shape=(2,), dtype=int64, numpy=array([0, 1], dtype=int64)>)
```
```

Reply:
```
for d in ('float32', 'float64', 'int32', 'uint8', 'int16', 
          'int8', 'complex64', 'int64', 'bfloat16', 'uint16', 
          'complex128', 'float16', 'uint32', 'uint64'): 
    x = tf.constant([1, 2,], dtype=d)
    print(cummax(x))        
    
(<tf.Tensor: shape=(2,), dtype=float32, numpy=array([1., 2.], dtype=float32)>, <tf.Tensor: shape=(2,), dtype=int64, numpy=array([0, 1], dtype=int64)>)
(<tf.Tensor: shape=(2,), dtype=float64, numpy=array([1., 2.])>, <tf.Tensor: shape=(2,), dtype=int64, numpy=array([0, 1], dtype=int64)>)
(<tf.Tensor: shape=(2,), dtype=int32, numpy=array([1, 2])>, <tf.Tensor: shape=(2,), dtype=int64, numpy=array([0, 1], dtype=int64)>)
(<tf.Tensor: shape=(2,), dtype=uint8, numpy=array([1, 2], dtype=uint8)>, <tf.Tensor: shape=(2,), dtype=int64, numpy=array([0, 1], dtype=int64)>)
(<tf.Tensor: shape=(2,), dtype=int16, numpy=array([1, 2], dtype=int16)>, <tf.Tensor: shape=(2,), dtype=int64, numpy=array([0, 1], dtype=int64)>)
(<tf.Tensor: shape=(2,), dtype=int8, numpy=array([1, 2], dtype=int8)>, <tf.Tensor: shape=(2,), dtype=int64, numpy=array([0, 1], dtype=int64)>)
(<tf.Tensor: shape=(2,), dtype=float32, numpy=array([1., 2.], dtype=float32)>, <tf.Tensor: shape=(2,), dtype=int64, numpy=array([0, 1], dtype=int64)>)
(<tf.Tensor: shape=(2,), dtype=int64, numpy=array([1, 2], dtype=int64)>, <tf.Tensor: shape=(2,), dtype=int64, numpy=array([0, 1], dtype=int64)>)
(<tf.Tensor: shape=(2,), dtype=bfloat16, numpy=array([1, 2], dtype=bfloat16)>, <tf.Tensor: shape=(2,), dtype=int64, numpy=array([0, 1], dtype=int64)>)
(<tf.Tensor: shape=(2,), dtype=uint16, numpy=array([1, 2], dtype=uint16)>, <tf.Tensor: shape=(2,), dtype=int64, numpy=array([0, 1], dtype=int64)>)
(<tf.Tensor: shape=(2,), dtype=float64, numpy=array([1., 2.])>, <tf.Tensor: shape=(2,), dtype=int64, numpy=array([0, 1], dtype=int64)>)
(<tf.Tensor: shape=(2,), dtype=float16, numpy=array([1., 2.], dtype=float16)>, <tf.Tensor: shape=(2,), dtype=int64, numpy=array([0, 1], dtype=int64)>)
(<tf.Tensor: shape=(2,), dtype=uint32, numpy=array([1, 2], dtype=uint32)>, <tf.Tensor: shape=(2,), dtype=int64, numpy=array([0, 1], dtype=int64)>)
(<tf.Tensor: shape=(2,), dtype=uint64, numpy=array([1, 2], dtype=uint64)>, <tf.Tensor: shape=(2,), dtype=int64, numpy=array([0, 1], dtype=int64)>)

```

Reply:
awesome, thanks for clarifying!